packages/@aws-cdk/aws-stepfunctions/lib/state-graph.ts

```diff
@@ -101,7 +102,7 @@ export class StateGraph {
     public toGraphJson(): object {
         const states: any = {};
         for (const state of this.allStates) {
-            states[state.stateId] = state.toStateJson();
+            states[state.stateId] = FieldUtils.renderObject(state.toStateJson());
```

Comment:
I feel like this should be the responsibility of the individual tasks. It only makes sense in the `Parameters` field, no?

Reply:
Yes, you are right. I forgot the reference path only applies for `Parameters` field, rather than the entire block of state.